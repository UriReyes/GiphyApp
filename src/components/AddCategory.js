import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmint = (e) => {
    e.preventDefault();
    if (inputValue !== undefined) {
      if (inputValue.trim().length > 2) {
        setCategories((cats) => [inputValue, ...cats]);
        setInputValue("");
      } else {
        alert("No has escrito nada");
      }
    }
  };

  return (
    <>
      <form className="formulario" onSubmit={handleSubmint}>
        <div className="d-flex-row">
          <input
            type="text"
            placeholder="Buscar"
            value={inputValue}
            onChange={handleInputChange}
          />
          <button
            className="d-flex-row align-items-center btn btn-search"
            onClick={handleSubmint}
          >
            <p>ðŸ”Ž</p>
          </button>
        </div>
      </form>
      <br />
    </>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
