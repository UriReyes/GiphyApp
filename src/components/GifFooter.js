import React from "react";

export const GifFooter = () => {
  return (
    <div className="footer">
      <div className="d-flex-row align-items-center justify-content-center">
        <h4 className="footer-app">Designed by UriCode &copy;</h4>
        <img
          className="img-footer"
          src={process.env.PUBLIC_URL + "helmet.svg"}
          alt="img_footer"
        />
      </div>
      <p className="text-center">
        <small>
          Este es un proyecto construido con el API de Giphy, todos los Gifs que
          aqu√≠ aparecen son pertenecientes a Giphy.
        </small>
      </p>
    </div>
  );
};
