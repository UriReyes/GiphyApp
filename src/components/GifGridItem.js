import React from "react";
import PropTypes from "prop-types";

import { useLoadingGif } from "../hooks/useLoadingGif";

export const GifGridItem = ({ title, url }) => {
  const gifData = useLoadingGif(url);

  const {
    event: imageLoaded,
    data: { gif, loaded },
  } = gifData;

  const showInformationHandler = (e) => {
    const cardInfo = e.currentTarget.firstChild;
    const cardInfoClasses = cardInfo.classList;

    if (cardInfoClasses.contains("d-none")) {
      cardInfoClasses.remove("d-none");
    }
  };

  const hideInformationHandler = (e) => {
    const cardInfo = e.currentTarget.firstChild;
    const cardInfoClasses = cardInfo.classList;

    if (!cardInfoClasses.contains("d-none")) {
      cardInfoClasses.add("d-none");
    }
  };

  return (
    <div
      className="d-flex justify-content-center flex-wrap card animate__animated animate__fadeIn n-father"
      onMouseOver={showInformationHandler}
      onMouseLeave={hideInformationHandler}
    >
      <div className="n-child d-none d-flex-row align-items-bottom justify-content-between">
        <p className="text-capitalize text-title card-info">{title}</p>
        <a
          href={url}
          className="btn card-info"
          target="_blank"
          rel="noreferrer"
        >
          ðŸ”—
        </a>
      </div>
      <img
        src={url}
        alt={title}
        onLoad={imageLoaded}
        srcSet={loaded ? gif : process.env.PUBLIC_URL + "loading.gif"}
      />
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
