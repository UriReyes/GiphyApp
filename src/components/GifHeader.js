import React from "react";
import PropTypes from "prop-types";

import { AddCategory } from "./AddCategory";

export const GifHeader = ({ setCategories }) => {
  return (
    <div className="nav d-flex align-items-center">
      <img
        src="https://media.giphy.com/media/3o6gbbuLW76jkt8vIc/giphy.gif"
        alt="img-header"
      />
      <div className="d-flex justify-content-center w-100">
        <AddCategory setCategories={setCategories} />
      </div>
    </div>
  );
};

GifHeader.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
