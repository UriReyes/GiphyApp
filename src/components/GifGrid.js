import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <div className="container text-center">
      <h3 className="text-capitalize">{category}</h3>
      <hr />
      <br />
      {loading && (
        <div className="lds-roller">
          <div></div>
          <div></div>
          <div></div>
          <div></div>
          <div></div>
          <div></div>
          <div></div>
          <div></div>
        </div>
      )}
      <div className="d-flex flex-wrap justify-content-between">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </div>
  );
};
