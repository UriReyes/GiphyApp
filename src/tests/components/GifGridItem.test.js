import React, { Component } from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";
import toJson from "enzyme-to-json";

describe("Probando GifGridItem", () => {
  const title = "Un titulo";
  const url = "https://imagen.gif";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Debe de mostrar el componente correctamente", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("Debe de contener una imagen igual al url y alt de los props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Debe div padre tener clases animate__fadeIn y n-father", () => {
    const div = wrapper.find("div");
    const divFather = div.at(0);

    expect(divFather.hasClass("animate__fadeIn n-father")).toBe(true);
  });

  test("Debe div hijo tener clases n-child", () => {
    const div = wrapper.find("div");
    const divChild = div.at(1);

    expect(divChild.hasClass("n-child")).toBe(true);
  });
});
