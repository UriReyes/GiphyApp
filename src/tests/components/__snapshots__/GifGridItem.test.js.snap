// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando GifGridItem Debe de mostrar el componente correctamente 1`] = `
<div
  className="d-flex justify-content-center flex-wrap card animate__animated animate__fadeIn n-father"
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="n-child d-none d-flex-row align-items-bottom justify-content-between"
  >
    <p
      className="text-capitalize text-title card-info"
    >
      Un titulo
    </p>
    <a
      className="btn card-info"
      href="https://imagen.gif"
      rel="noreferrer"
      target="_blank"
    >
      ðŸ”—
    </a>
  </div>
  <img
    alt="Un titulo"
    onLoad={[Function]}
    src="https://imagen.gif"
    srcSet="loading.gif"
  />
</div>
`;
